styx-common/src/main/java/com/spotify/styx/model/EventVisitor.java

```diff
@@ -37,7 +38,7 @@
 
   R triggerExecution(@Getter WorkflowInstance workflowInstance, Trigger trigger);
   R info(@Getter WorkflowInstance workflowInstance, Message message);
-  R dequeue(@Getter WorkflowInstance workflowInstance);
+  R dequeue(@Getter WorkflowInstance workflowInstance, @Nullable Set<String> resourceIds);
```

Comment:
Is it possible not to mark it as `@Nullable`, empty set should be good enough to indicate nothing.

Reply:
Done this way to be consistent with `submit` below, that similarly evolved a new field `executionId`.

Reply:
Oh, yeah, I didn't notice there was already two different prior approaches :/

Intellij actually highlights the `Optional<Integer>` used as type for parameter: "Optional was designed to provide a limited mechanism for library method _return types_"...